src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -664,6 +681,33 @@ private static bool ShowInStackTrace(MethodBase mb)
             return !(mb.IsDefined(typeof(StackTraceHiddenAttribute)) || (mb.DeclaringType?.IsDefined(typeof(StackTraceHiddenAttribute)) ?? false));
         }
 
+        private static bool TryResolveStateMachineMethod(ref MethodBase mb, ref Type declaringType)
```

Comment:
I don't find anything particularly wrong with it, but may I suggest inverting the if statement to reduce the nesting? I think it will help readability, but I would like to hear your option.

```csharp
private static bool TryResolveStateMachineMethod(ref MethodBase mb, ref Type declaringType)
{
    Debug.Assert(mb != null);
    Debug.Assert(declaringType != null);

    Type dt = declaringType.DeclaringType;
    if (dt == null) 
        return false;

    var methods = dt.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance | BindingFlags.DeclaredOnly);
    if(methods == null) 
        return false;

    foreach (var method in methods)
    {
        var asma = method.GetCustomAttribute<StateMachineAttribute>();
        if (asma != null && asma.StateMachineType == declaringType && method is MethodBase matched)
        {
            mb = matched;
            declaringType = matched.DeclaringType;
            return true;
        }
    }

    return false;
 }
```

Reply:
Reduced